kratos/processes/from_json_check_result_process.cpp

```diff
@@ -0,0 +1,866 @@
+    const int GPIndex
+    )
+{
+    if (ComponentIndex > -1) {
+        if (GPIndex > -1) {
+            KRATOS_WARNING("FromJSONCheckResultProcess") << "Error checking for variable " << rVariableName << " Component " << ComponentIndex << " in GP " << GPIndex << ", results:\n" << rEntityType << " " << EntityId << " " << std::setprecision(mRelevantDigits) << ValueEntity << "!=" << std::setprecision(mRelevantDigits) << ValueJSON << "; rel_tol=" << mRelativeTolerance << ", abs_tol=" << mAbsoluteTolerance << std::endl;
+        } else {
+            KRATOS_WARNING("FromJSONCheckResultProcess") << "Error checking for variable " << rVariableName << " Component " << ComponentIndex << ", results:\n" << rEntityType << " " << EntityId << " " << std::setprecision(mRelevantDigits) << ValueEntity << "!=" << std::setprecision(mRelevantDigits) << ValueJSON << "; rel_tol=" << mRelativeTolerance << ", abs_tol=" << mAbsoluteTolerance << std::endl;
+        }
+    } else {
+        if (GPIndex > -1) {
+            KRATOS_WARNING("FromJSONCheckResultProcess") << "Error checking for variable " << rVariableName << " in GP " << GPIndex << ", results:\n" << rEntityType << " " << EntityId << " " << std::setprecision(mRelevantDigits) << ValueEntity << "!=" << std::setprecision(mRelevantDigits) << ValueJSON << "; rel_tol=" << mRelativeTolerance << ", abs_tol=" << mAbsoluteTolerance << std::endl;
+        } else {
+            KRATOS_WARNING("FromJSONCheckResultProcess") << "Error checking for variable " << rVariableName << ", results:\n" << rEntityType << " " << EntityId << " " << std::setprecision(mRelevantDigits) << ValueEntity << "!=" << std::setprecision(mRelevantDigits) << ValueJSON << "; rel_tol=" << mRelativeTolerance << ", abs_tol=" << mAbsoluteTolerance << std::endl;
+        }
+    }
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+std::string FromJSONCheckResultProcess::GetNodeIdentifier(NodeType& rNode)
+{
+    if (this->Is(USE_NODE_COORDINATES)) {
+        const SizeType digits = 6;
+        std::stringstream ss;
+        ss.setf( std::ios::fixed );
+        ss << "X_" << std::setprecision( digits ) << rNode.X0() << "_Y_" << std::setprecision( digits ) << rNode.Y0() << "_Z_" << std::setprecision( digits ) << rNode.Z0();
+        return ss.str();
+    } else {
+        return std::to_string(rNode.Id());
+    }
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+FromJSONCheckResultProcess::NodesArrayType& FromJSONCheckResultProcess::GetNodes(const Flags* pFlag)
+{
+    KRATOS_TRY;
+
+    if (this->IsNot(NODES_CONTAINER_INITIALIZED)) {
+        auto& r_nodes_array = this->Is(CHECK_ONLY_LOCAL_ENTITIES) ? mrModelPart.GetCommunicator().LocalMesh().Nodes() : mrModelPart.Nodes();
```

Comment:
Why is this needed? I think you can always check the local nodes (the ghost ones would be check in their corresponding partitions).

Reply:
Thta's because retrocompatibility, I preer @philbucher to answer

Reply:
mmm... I don't understand why retrocompatibilty is broken (as far as I know the serial communicator has been always in there).

Reply:
It depends really if your code syncs the variables on the Nodes or not, both are valid use-cases, hence one can decide it
Obviously this only matters in MPI

Reply:
@philbucher should I keep it for retrocompatibility?

Reply:
This is needed! It cannot be removed, sorry I maybe explained myself badly